spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe "GET #show" do
+    let(:user) { create(:user) }
+
+    before :each do
+      sign_in user
+      get :show, params: { id: user.id }, format: :json
+    end
+    
+    it "returns the user information" do
+      user_response = JSON.parse(response.body, symbolize_names: true)
+      expect(user_response[:name]).to eq user.name
+      expect(user_response[:email]).to eq user.email
+      expect(user_response[:admin]).to eq user.admin
+    end
+  end
+
+  describe "PATCH #update" do
+    let(:user) { create(:user) }
+
+    context "when is succesfully updated" do
+      before :each do
+        sign_in user
+        patch :update, params: { id: user.id,
```

Comment:
don't use patch, use put, don't pass id as param, the user that should be updated is the user currently sign in